kebechet/exception.py

```diff
@@ -39,6 +39,30 @@ def __init__(self, command: Command, *args, **kwargs):
         self.raw_command = command
         super().__init__(*args)
 
+    def char_limit_dict(self, char_limit: int):
+        """Return command, stdout, and stderr in a dictionary with a maximum num. of characters allocated."""
+        to_ret = {}
+        attrs = [
+            (len(self.command), "command", self.command),
+            (len(self.stdout), "stdout", self.stdout),
+            (len(self.stderr), "stderr", self.stderr),
+        ]
+        attrs.sort()  # sorted according to string length, ensures largest attrs get highest char limit
+        individual_char_limit = int(char_limit / len(attrs))
+        for i in range(len(attrs)):
+            curr = attrs[i]
+            if curr[0] < individual_char_limit:
+                to_ret[curr[1]] = curr[2]
+
+                # distribute remaining limit, the max is to avoid divide by 0 on the final iteration
+                individual_char_limit += int(
+                    (individual_char_limit - curr[0]) / max(1, len(attrs) - i - 1)
+                )
+            else:
+                to_ret[curr[1]] = curr[2][: individual_char_limit - 3] + "..."
+
```

Comment:
thanks for adding this feature.
trying to make sense of this 

the loop would iterate for (command, std_out, std_err), can there be a case where command or std_err would take up all the `individual_char_limit`? if so then it seems one of the following options would take the most of the bits, we may have to change this a bit to modulate that.

Reply:
`individual_char_limit` is 1/3 of the total limit that way all will be able to use at least 1/3 and increases if the attributes do not use their allotted limit.

Reply:
oh right, I missed it was getting divided by the length.